pkg/bufferdecoder/eventsreader.go

```diff
@@ -367,34 +367,48 @@ func readStringFromBuff(ebpfMsgDecoder *EbpfDecoder) (string, error) {
 	return string(res), nil
 }
 
-// readStringVarFromBuff reads a null-terminated string from `buff`
-// max length can be passed as `max` to optimize memory allocation, otherwise pass 0
-func readStringVarFromBuff(decoder *EbpfDecoder, max int) (string, error) {
-	var err error
-	var char int8
-	res := make([]byte, 0, max)
-
-	err = decoder.DecodeInt8(&char)
-	if err != nil {
-		return "", errfmt.Errorf("error reading null terminated string: %v", err)
+// readSunPathFromBuff reads a sun_path string from the `buffer`.
+func readSunPathFromBuff(decoder *EbpfDecoder, size int) (string, error) {
+	if size <= 0 {
+		return "", errfmt.Errorf("size to decode sun_path must be greater than 0")
 	}
 
-	count := 1 // first char is already decoded
-	for char != 0 && count < max {
-		res = append(res, byte(char))
+	var err error
+	var char int8
+	res := make([]byte, 0, size)
 
-		// decode next char
+	count := 0
+	for i := 0; i < size; i++ {
 		err = decoder.DecodeInt8(&char)
 		if err != nil {
-			return "", errfmt.Errorf("error reading null terminated string: %v", err)
+			return "", errfmt.Errorf("error reading sun_path: %v", err)
 		}
 		count++
+
+		if char == 0 {
+			// char as NUL may signal the end of the string or an abstract socket
+			// https://elixir.bootlin.com/linux/v6.13.4/source/net/unix/af_unix.c#L72
+			// https://man7.org/linux/man-pages/man7/unix.7.html
+			if i > 0 {
+				// NUL found after the first char means the end of the string
+				break
+			}
+		}
+
+		res = append(res, byte(char))
+	}
+
+	if res[0] == 0 {
```

Comment:
I think this is what and the prior `if char == 0` is what specifies this logic. but the rest is still iterating over the variadic string. Could you maybe retain the original logic in its own routine (for potential future use), and wrap the sunpath logic around it? If we merge #4353 before you can probably use the SkipCursor method there for brevity, but manually doing so here would do just as well.

Reply:
Let's merge https://github.com/aquasecurity/tracee/pull/4353 first. So I can manage these changes accordingly. This fix can be part of v0.23.2.

Reply:
I don't think my PR is necessary, its just for clarity. Better that you make that refactor and I can just switch out into `MoveCursor` in my PR when I merge, which would serve as a usage example too.

Reply:
I've decided to keep the change, otherwise we should unroll two DecodeInt8 to make sure if it's either an empty or an abstract socket sunpath.

Reply:
I think this could be solved by introducing a `minimumLengthToDecode` parameter to the variadic string unwrapping. So the readSunPath function would pass `1` there to force read the first char, and then if it turns out to terminate afterwards you can make the relevant check you have in the next code segment.

Reply:
I might be a good option. I'm merging as approved. When we touch that again, let's recall to this. @geyslan @NDStrahilevitz @rscampos